tools/main.py

```diff
@@ -14,19 +15,93 @@
 
 import torchreid
-from torchreid.engine import build_engine
+from torchreid.engine import build_engine, get_initial_lr_from_checkpoint
 from torchreid.ops import DataParallel
 from torchreid.utils import (Logger, check_isfile, collect_env_info,
                              compute_model_complexity, resume_from_checkpoint,
                              set_random_seed, load_pretrained_weights)
 
+from torchreid.integration.nncf.compression import wrap_nncf_model, is_checkpoint_nncf
+
+def is_lr_set_from_opts(opts):
+    # Note that opts here -- the args.opts from build_base_argparser
+    key_names = opts[0::2]
+    return ('train.lr' in key_names)
+
+def calculate_lr_for_nncf_training(cfg, coeff_decrease_lr_for_nncf,
+                                   is_initial_lr_set_from_opts):
+    if is_initial_lr_set_from_opts:
+        print(f'Since initial LR is set from command line arguments, do not calculate initial LR for NNCF '
+              f'cfg.train.lr={cfg.train.lr}')
+        return cfg.train.lr
+
+    initial_lr_from_checkpoint = get_initial_lr_from_checkpoint(cfg.model.load_weights)
+    print(f'initial_lr_from_checkpoint = {initial_lr_from_checkpoint}')
+    if initial_lr_from_checkpoint is None:
+        print(f'The checkpoint does not contain initial lr -- will not calculate initial LR for NNCF, '
+              f'cfg.train.lr={cfg.train.lr}')
+        return cfg.train.lr
+
+    if is_checkpoint_nncf(cfg.model.load_weights):
+        print('The loaded checkpoint was received from NNCF training')
+        print('WARNING: Loading NNCF checkpoint for fine-tuning -- since LR is NOT set from command line arguments '
+              'the initial LR stored in checkpoint will be used again')
+        print('Please, set LR from command line arguments to avoid this behavior')
+        lr = initial_lr_from_checkpoint
+        print(f'lr = {lr}')
+        return lr
+
+    print('Try to calculate initial LR for NNCF')
+    print(f'coeff_decrease_lr_for_nncf = {coeff_decrease_lr_for_nncf}')
+    lr = initial_lr_from_checkpoint * coeff_decrease_lr_for_nncf
+    print(f'calculated lr = {lr}')
+    return lr
+
+def make_nncf_changes_in_training(model, cfg, classes, is_initial_lr_set_from_opts):
+    print(f'using NNCF')
+    lr = None
+    if cfg.model.resume:
+        raise NotImplementedError('Resuming NNCF training not implemented yet')
+    if not cfg.model.load_weights:
+        raise RuntimeError('NNCF training should be started from a non-NNCF or NNCF pre-trained model')
+    checkpoint_path = cfg.model.load_weights
+    if not check_isfile(checkpoint_path):
+        raise RuntimeError(f'Cannot load checkpoint from {checkpoint_path}')
+    if is_checkpoint_nncf(cfg.model.load_weights):
+        print(f'is_checkpoint_nncf = True')
+        # just skipping loading special datamanager
+        datamanager_for_nncf = None
+    else:
+        print('before building datamanager for nncf initializing')
```

Comment:
Sorry, but I think it is important.
1. The loading of dataset may require some time, so IMHO it is a good idea to inform a user what is making now.
2. Also (and it is even more important) when a datamanager is created, some logs will be printed to output. It may be confusing for users. IMHO it is a good idea to inform user why one more datamanager is created.

Reply:
NB: this function has been moved to `torchreid/integration/nncf/compression_script_utils.py`